Scutea Emilia/Lab 9/main.c

```diff
@@ -0,0 +1,108 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void extractSubstring(const char *string1[])
+{
+    int pos,nr;
+    printf("Input the position:\n");
+    scanf("%d",&pos);
+    printf("Input the number of characters:\n");
+    scanf("%d",&nr);
+    for(int i=pos; i<pos+nr; i++)
+        printf("%c",string1[i]);
+        printf("\n");
+
+}
+
+
+void insertString(const char *string1[],const char *string2[])
+{
+    int k,j=0;
+    printf("Input the position:\n");
+    scanf("%d",&k);
+    for(int i=strlen(string1)-1; i>=strlen(string1)-k; i--)
+    {
+        string1[i+strlen(string2)]=string1[i];
+    }
+    for(int i=k; i<=k+strlen(string2)-1; i++)
+    {
+        string1[i]=string2[j++];
+    }
+    puts(string1);
+}
+
+void deleteSubstring(const char *string1[])
+{
+    int pos,nr;
+    printf("Input the beginning position:\n");
+    scanf("%d",&pos);
+    printf("Input the length of the substring:\n");
+    scanf("%d",&nr);
+    for(int i=pos; i<=pos+nr; i++)
+        string1[i]=string1[i+nr];
+    for(int i=0; i<strlen(string1); i++)
+        printf("%c",string1[i]);
+        printf("\n");
+}
+
+void substringOfAnotherString(const char *string1[],const char *string2[])
+{
+    int n=strlen(string1);
+    int m=strlen(string2);
+    int j=0,ok=0,pos=0;
+    for(int i=0; i<n; i++)
+    {
+        if(string1[i]==string2[j])
+        {
+            pos=i;
+            i++;
+            j++;
+            while(string1[i]==string2[j])
+            {
+                i++;
+                j++;
+            }
+            if(j==m-1)
+            {
+                ok=1;
+                printf("The beginning position of the string is %d\n",pos);
+            }
+            else
+            {
+                j=0;
+            }
+
+        }
+    }
+    if(ok==0)
+    {
+        printf("The string is not a substring\n");
+    }
+}
+int main()
```

Comment:
The main function looks good but I'm pretty sure you did not understand pointers/arrays properly. Either ask me to explain them again at the lab or look over the examples I talked about in the videos. Also, I don't see the point of using `const` all that much in these examples.
Fyi, it's enough to write `char * string1 = (char*) malloc(sizeof(char)*50);` 
You don't need the extra `[]` - please pay more attention during the lab